docs/resolve/relationships.rst

```diff
@@ -142,50 +142,98 @@ This is by design because you wouldn't want one component to dispose the ``B`` o
 
 .. sourcecode:: csharp
+    public class B
+    {
+      public B(string someString, int id) {}
+      
+      public void DoSomething() {}
+    }
 
     public class A
     {
-        Func<int, string, B> _b;
+        Func<int, string, B> _newB;
 
-        public A(Func<int, string, B> b) { _b = b }
+        public A(Func<int, string, B> b) { _newB = b }
 
         public void M()
         {
-            var b = _b(42, "http://hel.owr.ld");
+            var b = _newB(42, "http://hell.owor.ld");
+            b.DoSomething();
+        }
+    }
+
+Note that since we're ``.Resolve()``ing the instantiation, rather than actually directly calling the constructor we don't need to declare the parameters in the same order as they appear in the ctor, nor do we necessarily need to provide *all* the parameters listed in that constructor. If some of the constructor's parameters could have been resolved by the Autofac Container already, then those parameters can be omitted from the ``Func`` signature being declared, so you only need to list the types that the Container can't resolve.
+
+Alternatively, you can use this approach to override a constructor parameter that *would* otherwise have been resolved from the container, with a concrete instance already in hand.
+
+Example:
+.. sourcecode:: csharp
+    //Suppose that P, Q & R are all registered with the Autofac Container.
+    public class B
+    {
+      public B(int id, P peaDependency, Q queueDependency, R ourDependency) {}
+      
+      public void DoSomething() {}
+    }
+
+    public class A
+    {
+        Func<int, P, B> _newB;
+
+        public A(Func<int, P, B> bFactory) { _newB = bFactory }
+
+        public void M(P existingPea)
+        {
+            var b = _newB(42, existingPea); //the needed Q and R will be resolved by Autofac, but not the P since the existingPea will be used.
             b.DoSomething();
         }
     }
 
-Internally, Autofac treats these as typed parameters. What that means is that **auto-generated function factories cannot have duplicate types in the input parameter list.** For example, say you have a type like this:
+Internally, Autofac determines what values to use for the constructor args, solely based on the types, and behaves as though we've temporarily defined the input values as the objects that should be used to resolve those types. A direct consequence of this is that  **auto-generated function factories cannot have duplicate types in the input parameter list.** See below for further notes on this.
```

Comment:
```suggestion
Internally, Autofac determines what values to use for the constructor args solely based on the type and behaves as though we've temporarily defined the input values for resolution. A consequence of this is that  **auto-generated function factories cannot have duplicate types in the input parameter list.** See below for further notes on this.
```

Reply:
Just to check, supposing we overode a type that could otherwise be resolved, but that type was also being resolved elsewhere in the dependency chain, as a child dependency of some other auto-resolved ctor parameter, what would happen?

i.e. imagine:

```
    class A
    {
        public A(B b, C c)
        {
            CapturedBee = b;
            CapturedSea = c;
        }
        public B CapturedBee;
        public C CapturedSea;
    }

    class B
    {
        public B(C c)
        {
            CapturedSubSea= c;
        }
        public C CapturedSubSea;
    }

    class Elsewhere
    {
        private Func<C, A> _newA;
        public A(Func<C, A> aFactory) { _newA = aFactory; }

        public void DoStuff(C existingSea)
        {
            var a = _newA(existingSea);

            var directResolution //true.
                = Object.ReferenceEquals(existingSea, a.CapturedSea);

            var indirectResolution // ???
                = Object.ReferenceEquals(existingSea, a.CapturedBee.CapturedSubSea);
        }
    }
```
What will `indirectResolution` be?

Reply:
There's a lot of "it depends" because we don't know where the `existingSea` in `DoStuff` came from and we don't know the lifetime scopes in which A, B, and C are registered. If they're all singletons the answer may be different than if they're all instance-per-dependency.

However, parameters are provided to the immediate component being resolved, _not to things down the resolution chain_. [That's actually one of our FAQs.](https://autofac.readthedocs.io/en/latest/faq/injecting-global-parameters.html) For all intents and purposes, the `existingSea` and `capturedSea` are likely not going to be the same. They wouldn't be in any case, whether you were calling `Resolve` directly with parameters or using `Func` relationships.

Reply:
Cool. So it's not that you're temporarily overriding the *general* rules for resolving that type. It's specifically overriding *how it should be resolved for the argument of this _particular_ ctor*